### ISSUE TITLE
Dashicons not enqueued in tablet/mobile preview if SCRIPT_DEBUG is true

### ISSUE THREAD

**Issue by bfintal** [WordPress/gutenberg#44205]
### Description

I used a dashicon inside my custom block that shows up only in the editor (for some type of UI in the editor). It shows up fine inside the content in the editor, however when I use the debug scripts by adding this in my `wp-config.php`...

```php
define( 'SCRIPT_DEBUG', true );
```

When I preview in tablet or mobile in the editor, the dashicons aren't visible anymore - as if the dashicon font file wasn't loaded or isn't available anymore. Without the `SCRIPT_DEBUG`, it works fine.

Expectation is that the behavior would be the same with or without `SCRIPT_DEBUG`.

### Step-by-step reproduction instructions

To reproduce this easily, I made a very simple custom block that just shows a dashicon in the editor:

1. Make sure that you have `define( 'SCRIPT_DEBUG', true );` in your **wp-config.php**
2. Create a new page
3. Add this custom block to your editor by pasting in your browser's console while in the editor:
```js
const Dashicon = wp.components.Dashicon
wp.blocks.registerBlockType( 'example/dashicon', {
	title: 'dashicon',
	edit: () => wp.element.createElement( wp.components.Dashicon, { icon: 'admin-site' } ),
	save: () => null,
} )
```
4. Add the example **dashicon** block
5. Click on **preview > tablet**
6. See that the dashicon icon disappears

**If you remove step 1, then you'll still see the dashicon in step 6.**

### Screenshots, screen recording, code snippet

Sample dashicon block to see the issue, paste in your browser console:

```js
const Dashicon = wp.components.Dashicon
wp.blocks.registerBlockType( 'example/dashicon', {
	title: 'dashicon',
	edit: () => wp.element.createElement( wp.components.Dashicon, { icon: 'admin-site' } ),
	save: () => null,
} )
```

Here is the normal tablet preview (the globe icon is the sample block):
![image](https://user-images.githubusercontent.com/1033611/190453591-6443d9a7-61d3-4111-90e2-b30f8576d59f.png)

Here is the tablet preview with `define( 'SCRIPT_DEBUG', true );`:
![image](https://user-images.githubusercontent.com/1033611/190454053-6a5cfdbd-bf0a-42b0-8a34-04e618318bc2.png)

### Environment info

- WordPress 6.0.2
- Brave [Version 1.27.109 Chromium: 92.0.4515.115 (Official Build) (x86_64)](https://brave.com/latest/)
- MacOS Big Sur 11.6.6
- (note that this is visible also in Chrome using the latest MacOS)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mikemcalister** [comment]
I can confirm this bug report. When `SCRIPT_DEBUG` is set to `TRUE`, Dashicons used in custom situations (blocks, patterns, etc.) the icon will not render (in my case, it shows as a blank square). However, with `SCRIPT_DEBUG` set to `FALSE`, the icon renders as expected.

**Comment by AnthonyLedesma** [comment]
This issue is also repeatable inside the `site-editor.php` screen. When viewing from `post.php` the block works but inside of `site-editor.php` with `SCRIPT_DEBUG` set to true we always get broken dashicons.

**SCRIPT_DEBUG=0**
![image](https://github.com/WordPress/gutenberg/assets/30462574/797d5199-e3c7-4ed4-be63-2d20272584b6)

**SCRIPT_DEBUG=1**
![image (1)](https://github.com/WordPress/gutenberg/assets/30462574/d2df66a0-1993-4361-85cb-923ae3c7f08a)

**Comment by Mamaduka** [comment]
I can't reproduce the issue when the Gutenberg plugin (15.8.1) is active. The plugin contains improvements for the iframed editor asset loading, so I think the issue was resolved there.

The fix should be part of WP 6.3.

**Comment by Mamaduka** [comment]
I'm going to close the issue since the bug is fixed in the plugin.
